enhancements/api-review/add-new-CRD-ImageDigestMirrorSet-and-ImageTagMirrorSet-to-config.openshift.io.md

```diff
@@ -0,0 +1,429 @@
+
+- The ImageContentSourcePolicy(ICSP) will follow the rules from [Deprecating an entire component](https://docs.openshift.com/container-platform/4.8/rest_api/understanding-api-support-tiers.html#deprecating-entire-component_understanding-api-tiers). The duration is
+12 months or 3 releases from the announcement of deprecation, whichever is longer.  
+
+### Upgrade / Downgrade Strategy
+
+#### Upgrade Strategy
+
+An existing cluster is required to make an upgrade to 4.11 in order to make use of the allow mirror by tags feature. Before the deprecation of ImageContentSourcePolicy, MCO needs to watch for both ImageContentSourcePolicy, and ImageDigestMirrorSet and create objects.
+
+Tier 1 component `imagecontentsourcepolicy.operator.openshift.io/v1alpha1` will stay 12 months or 3 releases from the announcement of deprecation.
+During the development on the release that one release ahead of deprecating `imagecontentsourcepolicy.operator.openshift.io/v1alpha1`,
+MCO will copy existing ImageContentSourcePolicy objects to ImageDigestMirrorSet and create new objects, and delete the
+ImageContentSourcePolicy objects. If any errors appear during the process, MCO should report `Upgradeable=False`.
+
+On the release that the ImageContentSourcePolicy CRD is removed from the API, the MCO will update its clusteroperator object to reflect a degraded state if it still finds objects of
+ImageContentSourcePolicy. The MCO should report that the ImageContentSourcePolicy is orphaned and let the user know they should create new objects
+using the new ImageDigestMirrorSet or ImageTagMirrorSet CRDs.
+
+#### Downgrade Strategy
+According to [Deprecating an entire component](https://docs.openshift.com/container-platform/4.8/rest_api/understanding-api-support-tiers.html#deprecating-entire-component_understanding-api-tiers), Tier 1 component
+`imagecontentsourcepolicy.operator.openshift.io/v1alpha1` will stay 12 months or 3 releases from the announcement of deprecation, whichever is longer. During the upgrade path,
+if an 4.11 upgrade fails mid-way through, or if the 4.11 cluster is
+misbehaving, the user can rollback to the version that supports `ImageContentSourcePolicy`.
+Their `ImageDigestMirrorSet` and `ImageTagMirrorSet` dependent workflows will be clobbered and broken if rollback to a version
+that lacks support for these CRDs. They can still configure the `ImageContentSourcePolicy` to use mirrors and keep previous behavior.
+
+### Version Skew Strategy
+
+Upgrade skew will not impact this feature. The MCO does not require skew check. CRI-O with n-2 OpenShift skew will still be able to handle the new property.
```

Comment:
Right now, c/image _does not_ implement tag-only mirrors; that will only land in the future. So, as of this moment, an n-2 CRI-O can’t process tag-only-mirror `registries.conf`. Do we need to worry about that?

The skew strategy section seems to be only intended for upgrades of older clusters, where we _can_ represent the feature using a configuration consumable by old CRI-O; presumably any new `*MirrorSet` CRs would only be added _after_ the upgrade — so we are probably fine, it’s just the “CRI-O with n-2 OpenShift skew … able to handle the new property” that seems to be claiming too much.